bt_ddos_shield/state_manager.py

```diff
@@ -0,0 +1,83 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from bt_ddos_shield.domain import Domain
+
+
+class State:
+    """
+    Class representing state of MinerShield.
+    """
+
+    class ShieldPhase(Enum):
+        """
+        Possible phases of shield.
+        """
+        DISABLED = "disabled"                         # disabled - initial state
+        MANIFEST_PUBLISHED = "manifest_published"     # manifest is saved to storage
+        MANIFEST_BROADCASTED = "manifest_broadcasted" # info about manifest is published to blockchain
+        ENABLED = "enabled"                           # shield is enabled
+
+    phase: ShieldPhase                # current phase of the shield
+    banned_validators: dict[str, int] # banned validators with ban time (HotKey -> ban timestamp)
```

Comment:
So the type is `HotKey` or `str`?
and then it's `int` or `datetime`?

Reply:
I made HotKey string, because maybe this type is declared in some library, which I haven't checked yet. But anyway I should add alias for this type and assume it is str as for now.

I thought about it as unix timestamp - I have bad experience with using datetime in MySQL databases, because of problems with time zone change and it will go into the database. But in this case it doesn't matter at all, so I will change it.